packages/mongodb-dialects/mongosh/src/main/kotlin/com/mongodb/jbplugin/dialects/mongosh/MongoshDialectFormatter.kt

```diff
@@ -172,6 +177,57 @@ private fun <S> MongoshBackend.emitQueryBody(
                     emitObjectValueEnd()
                 }
                 emitArrayEnd()
+                if (firstCall) {
+                    emitObjectEnd()
+                }
+            } else if (named.name == Name.NOT && hasChildren?.children?.size == 1) {
+                // the not operator is a special case
+                // because we receive it as:
+                // $not: { $field$: $condition$ }
+                // and it needs to be:
+                // $field$: { $not: $condition$ }
+                // we will do a JIT translation
+
+                var innerChild = hasChildren.children.first()
+                val operation = innerChild.component<Named>()
+                val fieldRef = innerChild.component<HasFieldReference<S>>()
+                val valueRef = innerChild.component<HasValueReference<S>>()
+
+                if (fieldRef == null) { // we are in an "and" / "or"...
+                    // so we use $nor instead
+                    emitQueryBody(
+                        Node(
+                            node.source,
+                            node.components<Component>().filterNot { it is Named } + Named(Name.NOR)
```

Comment:
Why do we emit Nor when it could also be inside an And?

Reply:
Because when you try to run a $not / $and the mongosh fails and suggests to use $nor for some reason, so I'm applying this automatically.

Reply:
Interesting - I was not aware of this behavior. Its also in the documentation:
```
[$nor](https://www.mongodb.com/docs/manual/reference/operator/query/nor/#mongodb-query-op.-nor) performs a logical NOR operation on an array of one or more query predicates and selects the documents that fail all the query predicates in the array
```
Thank you for the context :)